src/test/test-execute.c

```diff
@@ -209,11 +209,19 @@ static bool is_inaccessible_available(void) {
 static void _test(const char *file, unsigned line, const char *func,
                   Manager *m, const char *unit_name, int status_expected, int code_expected) {
         Unit *unit;
+        Unit *slice;
 
         assert_se(unit_name);
 
         assert_se(manager_load_startable_unit_or_warn(m, unit_name, NULL, &unit) >= 0);
         assert_se(unit_start(unit, NULL) >= 0);
+        slice = UNIT_GET_SLICE(unit);
+        if (slice) {
+                /* We need to start the slice as well otherwise the slice cgroup might be pruned
+                 * in on_cgroup_empty_event. */
+                int r = unit_start(slice, NULL);
+                assert_se(r >= 0 || r == -EALREADY);
+        }
```

Comment:
a slice should be started before any contained unit.

please move this up hence. and please make this recursive, just to be fully accurate, i.e. start all parent slice first, up the tree.